Sources/FluentKit/Migration/Migrator.swift

```diff
@@ -33,82 +33,130 @@ public struct Migrator {
+                self.lastBatchNumber(on: database)
+                    .and(value: migrations)
+            }.flatMap { (lastBatch, migrations) in
+                .andAllSync(migrations.map { item in
+                    { self.prepare(item, batch: lastBatch + 1) }
+                }, on: self.eventLoop)
+            }
         }
     }
     
     // MARK: Revert
     
-    public func revertLastBatch() -> EventLoopFuture<Void> {
-        self.lastBatchNumber().flatMap {
-            self.revertBatch(number: $0)
+    public func revertLastBatch(on databaseID: DatabaseID? = nil) -> EventLoopFuture<Void> {
+        self.run(on: databaseID) { database in
+            self.lastBatchNumber(on: database).flatMap {
+                self.revertBatch(number: $0, on: database)
+            }
         }
     }
     
-    public func revertBatch(number: Int) -> EventLoopFuture<Void> {
-        self.preparedMigrations(batch: number).flatMap { migrations in
-            EventLoopFuture<Void>.andAllSync(migrations.map { item in
-                { self.revert(item) }
-            }, on: self.eventLoop)
+    public func revertBatch(number: Int, on databaseID: DatabaseID? = nil) -> EventLoopFuture<Void> {
+        self.run(on: databaseID) { database in
+            self.preparedMigrations(batch: number, on: database).flatMap { migrations in
+                EventLoopFuture<Void>.andAllSync(migrations.map { item in
+                    { self.revert(item) }
+                }, on: self.eventLoop)
+            }
         }
     }
     
-    public func revertAllBatches() -> EventLoopFuture<Void> {
-        self.preparedMigrations().flatMap { migrations in
-            .andAllSync(migrations.map { item in
-                { self.revert(item) }
-            }, on: self.eventLoop)
-        }.flatMap { _ in
-            self.revertMigrationLog()
+    public func revertAllBatches(on databaseID: DatabaseID? = nil) -> EventLoopFuture<Void> {
+        self.run(on: databaseID) { database in
+            self.preparedMigrations(on: database).flatMap { migrations in
+                .andAllSync(migrations.map { item in
+                    { self.revert(item) }
+                }, on: self.eventLoop)
+            }.flatMap { _ in
+                self.revertMigrationLog(on: database)
+            }
         }
     }
     
     // MARK: Preview
     
-    public func previewPrepareBatch() -> EventLoopFuture<[(Migration, DatabaseID?)]> {
-        self.unpreparedMigrations().map { items in
-            items.map { item  in
-                (item.migration, item.id)
+    public func previewPrepareBatch(on databaseID: DatabaseID? = nil) -> EventLoopFuture<[(Migration, DatabaseID?)]> {
+        var batch: [(Migration, DatabaseID?)] = []
+        var failed: Error? = nil
+
+        return self.run(on: databaseID) { database in
+            if case .some = failed { return self.eventLoop.makeSucceededFuture(()) }
```

Comment:
You can just write `guard failed == nil else` for this, you don't need the awful backwards enum pattern match syntax 🙂

Reply:
@calebkleveter Looks like you  missed one